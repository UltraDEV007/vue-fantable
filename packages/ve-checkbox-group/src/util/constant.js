// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
