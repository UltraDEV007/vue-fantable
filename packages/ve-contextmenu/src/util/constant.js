// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
