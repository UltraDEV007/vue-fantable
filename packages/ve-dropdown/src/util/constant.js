// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
