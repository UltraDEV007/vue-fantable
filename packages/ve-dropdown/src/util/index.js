/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
