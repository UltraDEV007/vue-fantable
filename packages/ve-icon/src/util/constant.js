// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
