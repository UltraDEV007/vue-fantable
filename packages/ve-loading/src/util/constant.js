// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
