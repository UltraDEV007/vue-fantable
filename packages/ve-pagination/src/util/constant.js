// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
