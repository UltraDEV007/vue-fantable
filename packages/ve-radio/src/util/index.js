import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
