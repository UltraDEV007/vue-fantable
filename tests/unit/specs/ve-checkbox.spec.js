import { mount } from "@vue/test-utils";
import veCheckbox from "@/ve-checkbox";
import { later } from "../util";

describe("veCheckbox", () => {
    it("render normal", () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                value: true,
                label: "orange",
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render with checkbox group", async () => {
        const wrapper = mount({
            template: `
                <ve-checkbox-group v-model="checkboxValue">
                    <ve-checkbox label="南瓜" />
                    <ve-checkbox disabled label="西红柿" />
                    <ve-checkbox label="哈密瓜" />
                    <ve-checkbox label="水蜜桃" />
                </ve-checkbox-group>
            `,
            data() {
                return {
                    checkboxValue: ["西红柿", "哈密瓜"],
                };
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render with indeterminate", () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                value: true,
                label: "orange",
                disabled: true,
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render with disabled", () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                value: true,
                label: "orange",
                indeterminate: true,
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("value prop", () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                value: true,
                label: "orange",
            },
        });

        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(true);
    });

    it("label prop", () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                value: true,
                label: "orange",
            },
        });

        expect(wrapper.find(".ve-checkbox-label").text()).toBe("orange");
    });

    it("label content slot", () => {
        const wrapper = mount(veCheckbox, {
            slots: {
                default: "orange",
            },
        });

        expect(wrapper.find(".ve-checkbox-label").text()).toBe("orange");
    });

    it("disbled prop", () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                disabled: true,
                label: "orange",
            },
        });

        expect(wrapper.find(".ve-checkbox-disabled").exists()).toBe(true);
    });

    it("disbled selected with checked statue", async () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                value: true,
                disabled: true,
                label: "orange",
            },
        });
        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(true);

        wrapper.find(".ve-checkbox").trigger("click");
        await later();
        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(true);
    });

    it("disbled selected with unchecked statue", async () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                value: false,
                disabled: true,
                label: "orange",
            },
        });
        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(false);

        wrapper.find(".ve-checkbox").trigger("click");
        await later();
        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(false);
    });

    it("indeterminate prop", () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                indeterminate: true,
                label: "orange",
            },
        });
        expect(wrapper.find(".ve-checkbox-indeterminate").exists()).toBe(true);
    });

    it("isControlled prop", async () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                isControlled: true,
                label: "orange",
            },
        });

        await wrapper.setProps({ isSelected: true });
        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(true);

        await wrapper.setProps({ isSelected: false });
        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(false);
    });

    it("isSelected prop", () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                isControlled: true,
                isSelected: true,
                label: "orange",
            },
        });

        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(true);
    });

    it("click event", async () => {
        const wrapper = mount({
            template: `
            <ve-checkbox v-model="checkboxValue">orange</ve-checkbox>`,
            data() {
                return {
                    checkboxValue: false,
                };
            },
        });

        wrapper.find(".ve-checkbox").trigger("click");
        await later();
        expect(wrapper.find(".ve-checkbox-checked").exists()).toBe(true);
    });

    it("checked change emit", async () => {
        const wrapper = mount(veCheckbox, {
            propsData: {
                value: false,
                label: "orange",
            },
        });

        wrapper.find(".ve-checkbox").trigger("click");
        wrapper.find(".ve-checkbox").trigger("click");

        await later();

        expect(wrapper.emitted("on-checked-change").length).toEqual(2);
        expect(wrapper.emitted("on-checked-change")[0]).toEqual([true]);
        expect(wrapper.emitted("on-checked-change")[1]).toEqual([false]);
    });

    it("checkboxGroup data change", async () => {
        const wrapper = mount({
            template: `
                <ve-checkbox-group v-model="checkboxValue">
                    <ve-checkbox label="南瓜" />
                    <ve-checkbox disabled label="西红柿" />
                    <ve-checkbox label="哈密瓜" />
                    <ve-checkbox label="水蜜桃" />
                </ve-checkbox-group>
            `,
            data() {
                return {
                    checkboxValue: ["西红柿", "哈密瓜"],
                };
            },
        });

        const checkeds1 = wrapper.findAll(".ve-checkbox-checked");

        expect(checkeds1.length).toBe(2);

        wrapper.setData({ checkboxValue: ["西红柿"] });

        await later();

        const checkeds2 = wrapper.findAll(".ve-checkbox-checked");

        expect(checkeds2.length).toBe(1);

        wrapper.setData({ checkboxValue: ["西红柿", "水蜜桃"] });

        await later();

        const checkeds3 = wrapper.findAll(".ve-checkbox-checked");

        expect(checkeds3.length).toBe(2);
    });

    it("checkbox data change width checkboxGroup", async () => {
        const wrapper = mount({
            template: `
                <ve-checkbox-group v-model="checkboxValue">
                    <ve-checkbox label="南瓜" />
                    <ve-checkbox disabled label="西红柿" />
                    <ve-checkbox label="哈密瓜" />
                    <ve-checkbox label="水蜜桃" />
                </ve-checkbox-group>
            `,
            data() {
                return {
                    checkboxValue: ["西红柿", "哈密瓜"],
                };
            },
        });

        wrapper.find(".ve-checkbox").trigger("click");

        await later();

        expect(wrapper.vm.checkboxValue).toEqual(["西红柿", "哈密瓜", "南瓜"]);
    });
});
