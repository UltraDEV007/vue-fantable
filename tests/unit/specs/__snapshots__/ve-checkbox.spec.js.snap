// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veCheckbox render normal 1`] = `<label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content ve-checkbox-checked"><input type="checkbox" class="ve-checkbox-input" value="orange"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">orange</span></label>`;

exports[`veCheckbox render with checkbox group 1`] = `<div class="ve-checkbox-group"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value="南瓜"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">南瓜</span></label> <label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content ve-checkbox-checked ve-checkbox-disabled"><input type="checkbox" class="ve-checkbox-input" value="西红柿"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">西红柿</span></label> <label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content ve-checkbox-checked"><input type="checkbox" class="ve-checkbox-input" value="哈密瓜"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">哈密瓜</span></label> <label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value="水蜜桃"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">水蜜桃</span></label></div>`;

exports[`veCheckbox render with disabled 1`] = `<label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content ve-checkbox-checked ve-checkbox-indeterminate"><input type="checkbox" class="ve-checkbox-input" value="orange"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">orange</span></label>`;

exports[`veCheckbox render with indeterminate 1`] = `<label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content ve-checkbox-checked ve-checkbox-disabled"><input type="checkbox" class="ve-checkbox-input" value="orange"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">orange</span></label>`;
